### PR TITLE
Helper method fixes Issue#671 that validates user ID

### PR DESCRIPTION
**Pull request by bquach2**
Implemented helper method requested in Issue https://github.com/cs-discord-at-ucf/lion/issues/671 and created and modified tests to test this method. This helper method takes in as input a string and uses regex to verify that the input string matches the discord user ID pattern. Fixes https://github.com/cs-discord-at-ucf/lion/issues/671

### CODE REVIEW

File: src/services/moderation.service.ts [hunk 1]
```diff
@@ -385,9 +389,9 @@ export class ModService {
   private async _kickUser(member: GuildMember) {
     await member.send(
       'You are being kicked for too many warnings\n' +
-        `You currently have been warned ${this._KICK_THRESH} times.\n` +
-        `After ${this._SUSPEND_THRESH} warnings, you will have restricted access to the server.\n` +
-        `After ${this._BAN_THRESH} warnings, you will be banned permanently.`
+      `You currently have been warned ${this._KICK_THRESH} times.\n` +
+      `After ${this._SUSPEND_THRESH} warnings, you will have restricted access to the server.\n` +
+      `After ${this._BAN_THRESH} warnings, you will be banned permanently.`
```
Comment (tanndlin) [inline src/services/moderation.service.ts:394]: There's quite of few unrelated formatting changes in this. The prettier vsc plugin has all of our styling if you don't have that installed. Or just try to remove these changes

**Review by tanndlin — CHANGES_REQUESTED** [review]
Looks good, but rename the name of the test file now, shouldn't be `checkclasses` anymore

**Review by c650 — APPROVED** [review]
Minor nit but lgtm. Why did you open and close three PR’s for this? 

Also please write “fixes #671” in the desc

**Review by tanndlin — CHANGES_REQUESTED** [review]
Sorry I got busy and forgot about this PR.
It would also make sense to implement this in the `check_classes.plugin.ts` where it originally came from, that way its not unused code we forget to implement :)

**Comment by bquach2** [comment]
> Sorry I got busy and forgot about this PR. It would also make sense to implement this in the `check_classes.plugin.ts` where it originally came from, that way its not unused code we forget to implement :)

Just made the additions to the checkclasses plugin, thanks for the comment.

File: src/app/plugins/checkclasses.plugin.ts [hunk 2]
```diff
@@ -26,11 +26,15 @@ export default class CheckClassesPlugin extends Plugin {
...
-
     if (!member) {
       await message.reply('User not found.');
       return;
     }
+    const valid = Moderation.Helpers.validateUser(member.id);
+    if (!valid) {
+      await message.reply('User ID not valid.');
+      return;
+    }
```
Comment (tanndlin) [inline src/app/plugins/checkclasses.plugin.ts:37]: We have a built in validation method that you can override that gets called to see if the input is valid before the `execute` method is ran.
Look [here](https://github.com/cs-discord-at-ucf/lion/blob/eb3d25314da00e7474ddda6a7e7fae643d050576/src/app/plugins/leaderboard.plugin.ts#L16-L18) for an example.

When a message is sent, if its a command the `command handlers` calls [validate](https://github.com/cs-discord-at-ucf/lion/blob/eb3d25314da00e7474ddda6a7e7fae643d050576/src/app/handlers/command.handler.ts#L133-L136), and if that returns true, it runs execute. That way we have a standard way of validating args, and have a standardized error message.

**Review by tanndlin — CHANGES_REQUESTED** [review]

File: src/app/plugins/checkclasses.plugin.ts [hunk 3]
```diff
@@ -15,10 +15,17 @@ export default class CheckClassesPlugin extends Plugin {
 
   private _MAX_CHANS_SHOWN: number = 10;
 
+  public override validate(message: IMessage, args: string[]) {
+    const valid = Moderation.Helpers.validateUser(args[0]);
+    if (!valid) {
+      return false;
+    }
+    return true;
```
Comment (tanndlin) [inline src/app/plugins/checkclasses.plugin.ts:23]: If `args.length` == 0 then this will crash, need to check for that

Also, just `return Moderation.Helpers.validateUser(args[0]);`

**Review by tanndlin — CHANGES_REQUESTED** [review]

File: src/app/plugins/checkclasses.plugin.ts [hunk 4]
```diff
@@ -15,18 +15,24 @@ export default class CheckClassesPlugin extends Plugin {
...
```
Comment (c650) [inline src/app/plugins/checkclasses.plugin.ts:23]:
```
return !!Moderation.Helpers...
```

File: src/app/plugins/checkclasses.plugin.ts [hunk 5]
```diff
@@ -15,18 +15,24 @@ export default class CheckClassesPlugin extends Plugin {
...
+  }
   constructor(public container: IContainer) {
     super();
   }
-
+
```
Comment (c650) [inline src/app/plugins/checkclasses.plugin.ts:28]: pls configure your editor to not leave trailing spaces.

File: src/app/plugins/checkclasses.plugin.ts [hunk 6]
```diff
@@ -15,18 +15,24 @@ export default class CheckClassesPlugin extends Plugin {
...
-
+ 
   public async execute(message: IMessage, args: string[]) {
     const userHandle = args.join(' ');
 
     const member = await Moderation.Helpers.resolveUser(
       this.container.guildService.get(),
       userHandle
     );
-
```
Comment (c650) [inline src/app/plugins/checkclasses.plugin.ts:29]: undo

**Review by c650 — CHANGES_REQUESTED** [review]

**Review by tanndlin — APPROVED** [review]

File: src/app/plugins/checkclasses.plugin.ts (hunk 3 above)
Comment (tanndlin) [reply src/app/plugins/checkclasses.plugin.ts]: bump

If there are no args, calling `args[0]` will crash

File: src/app/plugins/checkclasses.plugin.ts [hunk 7]
```diff
@@ -15,6 +15,10 @@ export default class CheckClassesPlugin extends Plugin {
 
   private _MAX_CHANS_SHOWN: number = 10;
 
+  public override validate(message: IMessage, args: string[]) {
+    return !!Moderation.Helpers.validateUser(args[0]);
```
Comment (tanndlin) [inline src/app/plugins/checkclasses.plugin.ts:19]: Bumping this again.
Need to check args.length
If no args are given args[0] will crash

**Review by tanndlin — CHANGES_REQUESTED** [review]

File: src/app/plugins/checkclasses.plugin.ts (hunk 7 above)
Comment (bquach2) [reply src/app/plugins/checkclasses.plugin.ts]: Oh sorry, I thought the !! took care of that. Added this check in, thanks!

**Review by tanndlin — APPROVED** [review]

**Comment by bquach2** [comment]
@c650 Hi, when you have the time could you review my request please? This is my first pull request and I'm looking forward to having it accepted

File: src/app/plugins/checkclasses.plugin.ts [hunk 8]
```diff
@@ -15,6 +15,10 @@ export default class CheckClassesPlugin extends Plugin {
 
   private _MAX_CHANS_SHOWN: number = 10;
 
+  public override validate(message: IMessage, args: string[]) {
+    return args.length != 0 && !!Moderation.Helpers.validateUser(args[0]);
```
Comment (c650) [inline src/app/plugins/checkclasses.plugin.ts:19]: why are you !! On a function call that returns Boolean

Comment (tanndlin) [reply src/app/plugins/checkclasses.plugin.ts:19]: lol you did tell them to do that, but yeah it should be removed, good catch kek

**Comment by bquach2** [comment]
@c650 Yeah was just going off of what you said. Thought it made sense too at the time. I went ahead and adjusted it, thanks!

**Review by c650 — APPROVED** [review]

**Comment by tanndlin** [comment]
@bquach2 You should be able to merge whenever you want now that it has 2 approvals

**Comment by bquach2** [comment]
@tanndlin Hey it is saying that merging is restricted to authorized users and that I am unable to merge this pull request.

**Comment by tanndlin** [comment]
Guess i never set that up correctly lmao
